datafusion/physical-plan/src/windows/mod.rs

```diff
@@ -395,6 +379,8 @@ impl BuiltInWindowFunctionExpr for WindowUDFExpr {
     }
 }
 
+// TODO: Find a way to make clippy happy
+#[allow(clippy::needless_borrow)]
```

Comment:
I wonder why we get this warning now. Is it because we used to clone with `expr` and now do it with `&expr`? We should understand if this is a spurious warning or indicative of a code smell.

Reply:
```
error: this expression creates a reference which is immediately dereferenced by the compiler
   --> datafusion/physical-plan/src/windows/mod.rs:401:28
    |
401 |                 Arc::clone(&expr),
    |                            ^^^^^ help: change this to: `expr`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
    = note: `-D clippy::needless-borrow` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::needless_borrow)]`
```

This is the clippy error

Reply:
And I presume if you just change to `expr`, like how it was before, we get some other error?

Reply:
Interesting, actually it has no error. The error I got is from rust-analyzer.

`expected &Arc<dyn PhysicalExpr, Global>, found Arc<dyn PhysicalExpr, Global>`

😕